### PR TITLE
Fix warnings in the project

### PR DESCRIPTION
**Pull request by rowwingman**
### Short description 📝

Provide install with [Mint](https://github.com/yonaskolb/Mint/blob/master/README.md#support) wrapper around swift package manager.

### Solution 📦

Need to fix tuist project: 
* remove all warnings, so zero warnings policy
* refactor unit tests that are using internal API.

### CODE REVIEW

**Comment by rowwingman** [comment]
I have some errors while I build `tuist` package for release `swift build -c release`
```
tuist/Sources/TuistSupportTesting/Extensions/ArgumentParser+TestData.swift:2:18: error: module 'SPMUtility' was not compiled for testing
@testable import SPMUtility
```
```
tuist/Sources/TuistSupportTesting/TestCase/TuistTestCase.swift:5:18: error: module 'TuistSupport' was not compiled for testing
@testable import TuistSupport
```

Maybe someone can help me resolve `Tests`?

File: README.md [hunk 1]
```diff
@@ -64,12 +64,22 @@ The list of actions will likely grow as we get feedback from you.
...
-
+### Running script (Recommended):
 ```bash
 bash <(curl -Ls https://install.tuist.io)
 ```
 
+
+### Using [Homebrew](http://brew.sh/):
+```
+brew install tuist
```
Comment (kwridan) [inline README.md]: Is there a formula for tuist on home-brew core?

File: README.md [hunk 2]
```diff
@@ -64,12 +64,22 @@ The list of actions will likely grow as we get feedback from you.
...
+```
+
+### Using [Mint](https://github.com/yonaskolb/mint)
+```
+$ mint install tuist/tuist
```
Comment (kwridan) [inline README.md]: This was attempted in the past:
https://github.com/tuist/tuist/issues/315

Sadly it won't work out of the box due to tuist requiring an additional dynamic library `ProjectDescription` to be installed alongside it and if I recall correctly Mint only installs the executables?

File: Tests/TuistGeneratorTests/Settings/DefaultSettingsProviderTests.swift [hunk 3]
```diff
@@ -499,7 +499,10 @@ private extension XCTestCase {
                            file: StaticString = #file,
                            line: UInt = #line) {
         for (key, expectedValue) in second {
```
Comment (kwridan) [inline Tests/TuistGeneratorTests/Settings/DefaultSettingsProviderTests.swift]: for this particular test `nil` is a valid value to compare against, the warning is generated due to the string interpolation.

Perhaps to mitigate this warning, we can either do `String(describing:)` or `let resultDescription = results.map { "\($0)" } ?? "nil"` and use that in the `"\(key):\(resultDescription) does not match....`

File: README.md (hunk 2 above)
Comment (rowwingman) [reply README.md]: yeah, you are right, I've got. So I will change pull request to "Fix warnings"

File: README.md (hunk 1 above)
Comment (rowwingman) [reply README.md]: Removed this part

File: Tests/TuistGeneratorTests/Settings/DefaultSettingsProviderTests.swift (hunk 3 above)
Comment (rowwingman) [reply Tests/TuistGeneratorTests/Settings/DefaultSettingsProviderTests.swift]: thanks

**Comment by rowwingman** [comment]
Remove Mint integration. Leave only fix warnings

File: Sources/TuistSupport/Extensions/AbsolutePath+Extras.swift [hunk 4]
```diff
@@ -59,45 +58,7 @@ extension AbsolutePath {
...
-            digest.withUnsafeMutableBytes {
-                _ = CC_SHA256_Final($0, &context)
-            }
-
-            return digest
-        } catch {
-            print(error)
-            return nil
-        }
+        return try? SHA256Digest.file(at: url)
```
Comment (kwridan) [inline Sources/TuistSupport/Extensions/AbsolutePath+Extras.swift]: Just spotted there's already a `SHA256` utility in the `Basic` library which could serve as an alternate

Comment (rowwingman) [reply Sources/TuistSupport/Extensions/AbsolutePath+Extras.swift]: I added https://inneka.com/programming/swift/sha256-in-swift/ to the SHA256Diegest, as I see, implementation that I add is for files, there is a buffer of 4k bytes. Basic has an implementation for `Data`, so you read the all data and then generate sha. So, I think using input stream is more efficient.

**Comment by pepibumur** [comment]
Hey @rowwingman. Thanks for tackling those warnings in the project. I will revert the instructions for Homebrew & Mint because Tuist was designed to be installed using its own script and utilities for that. The version management is entirely done by `tuistenv`.

**Comment by rowwingman** [comment]
> I will revert the instructions for Homebrew & Mint because Tuist was designed to be installed using its own script and utilities for that. The version management is entirely done by `tuistenv`.
@pepibumur I reverted it yesterday, sorry about missing it

**Comment by pepibumur** [comment]
>  @pepibumur I reverted it yesterday, sorry about missing it

:+1: Sorry, I just saw the PR after I posted the comment 😬

**Pull request merged**